src/test/java/picard/sam/FilterReadsByFlowCellLocationTest.java

```diff
@@ -0,0 +1,163 @@
+package picard.sam;
+
+import htsjdk.samtools.*;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Unit tests for FilterFlowCellEdgeReads using TestNG.
+ */
+public class FilterReadsByFlowCellLocationTest {
+
+    // Temporary files for input and output.
+    private File inputSam;
+    private File outputSam;
+
+    /**
+     * Helper method to create a temporary SAM file with one record per provided read name.
+     * Each record is given minimal required fields.
+     *
+     * @param readNames an array of read names to include.
+     * @return the temporary SAM file.
+     * @throws IOException if an I/O error occurs.
+     */
+    private File createSamFile(String[] readNames) throws IOException {
+        File tmpSam = File.createTempFile("FilterFlowCellEdgeReadsTest_input", ".sam");
+        tmpSam.deleteOnExit();
+
+        // Create a minimal SAM file header.
+        SAMFileHeader header = new SAMFileHeader();
+        header.setSortOrder(SAMFileHeader.SortOrder.unsorted);
+        // Add one sequence record so that records have a reference.
+        header.addSequence(new SAMSequenceRecord("chr1", 1000000));
+
+        // Use SAMFileWriterFactory to write a SAM file.
+        try (SAMFileWriter writer = new SAMFileWriterFactory().makeWriter(header, false, tmpSam, null)) {
+            // Create one record for each read name.
+            for (String readName : readNames) {
+                SAMRecord rec = new SAMRecord(header);
+                rec.setReadName(readName);
+                rec.setReferenceName("chr1");
+                rec.setAlignmentStart(1);
+                rec.setCigarString("50M");
+                // Set dummy bases and qualities.
+                rec.setReadString("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA");
+                rec.setBaseQualityString("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF");
+                writer.addAlignment(rec);
+            }
+        }
+        return tmpSam;
+    }
+
+    /**
+     * Helper method to count the number of SAM records in a given SAM file.
+     *
+     * @param samFile the SAM file.
+     * @return the number of records.
+     * @throws IOException if an I/O error occurs.
+     */
+    private int countRecords(File samFile) throws IOException {
+        int count = 0;
+        try (SamReader reader = SamReaderFactory.makeDefault().open(samFile)) {
+            for (SAMRecord rec : reader) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    @AfterMethod
+    public void tearDown() {
```

Comment:
don't think you need this, because I think you can keep the inputSam and outputSam objects local to the tests instead of as class fields